pkg/updater/updater.go

```diff
@@ -1419,6 +1422,42 @@ func alertInfo(failures int32, msg, cellID, latestCellID string, userProperties
 	}
 }
 
+func columnStats(cells map[string]Cell, brokenThreshold float32) *statepb.Stats {
+	var passes, fails, total int32
+	var pending bool
+	if brokenThreshold <= 0.0 {
+		return nil
+	}
+	if cells == nil {
+		return nil
+	}
+	for _, cell := range cells {
+		if cell.Result == statuspb.TestStatus_RUNNING {
+			pending = true
+		}
+		status := result.Coalesce(cell.Result, false)
+		switch status {
+		case statuspb.TestStatus_PASS:
+			passes++
+			total++
+		case statuspb.TestStatus_FAIL:
+			fails++
+			total++
+		case statuspb.TestStatus_FLAKY, statuspb.TestStatus_UNKNOWN:
+			total++
+		default:
+			// blank cell or unrecognized status, do nothing
+		}
+	}
+	return &statepb.Stats{
+		FailCount:  fails,
+		PassCount:  passes,
+		TotalCount: total,
+		Pending:    pending,
+		Broken:     float32(fails)/float32(total) > brokenThreshold,
```

Comment:
if total is 0 this will blow up, for example

Reply:
Daaaaaangit.